File: src/main/scala/com/github/mauricio/postgresql/parsers/ReadyForQueryParser.scala
package com.github.mauricio.postgresql.parsers

import org.jboss.netty.buffer.ChannelBuffer
import com.github.mauricio.postgresql.messages.backend.{ReadyForQueryMessage, Message}

/**
 * User: Maurício Linhares
 * Date: 2/29/12
 * Time: 12:33 AM
 */

object ReadyForQueryParser extends MessageParser {

  override def parseMessage(b: ChannelBuffer): Message = {
    new ReadyForQueryMessage( b.readByte().asInstanceOf[Char] )
  }

}

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/parsers/ReadyForQueryParser.kt
package com.github.mauricio.async.db.postgresql.parsers

import com.github.mauricio.async.db.postgresql.messages.backend.ReadyForQueryMessage
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage
import io.netty.buffer.ByteBuf


object ReadyForQueryParser : MessageParser {

  override fun parseMessage(b: ByteBuf): ServerMessage {
    return ReadyForQueryMessage(b.readByte().toChar())
  }

}
